src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -680,5 +931,106 @@ internal int VerifyCallback(int ok, IntPtr ctx)
                 }
             }
         }
+
+        private unsafe struct ErrorCollection
+        {
+            // As of OpenSSL 1.1.1 there are 74 defined X509_V_ERR values.
+            private const int BucketCount = 3;
+            private fixed int _codes[BucketCount];
+
+            internal bool HasErrors =>
+                _codes[0] != 0 || _codes[1] != 0 || _codes[2] != 0;
+
+            internal void Add(Interop.Crypto.X509VerifyStatusCode statusCode)
+            {
+                int bucket = FindBucket(statusCode, out int bitValue);
+                _codes[bucket] |= bitValue;
+            }
+
+            public void ClearError(Interop.Crypto.X509VerifyStatusCode statusCode)
+            {
+                int bucket = FindBucket(statusCode, out int bitValue);
+                _codes[bucket] &= ~bitValue;
+            }
+
+            internal bool HasError(Interop.Crypto.X509VerifyStatusCode statusCode)
+            {
+                int bucket = FindBucket(statusCode, out int bitValue);
+                return (_codes[bucket] & bitValue) != 0;
+            }
+
+            public Enumerator GetEnumerator()
+            {
+                return new Enumerator(this);
+            }
+
+            private static int FindBucket(Interop.Crypto.X509VerifyStatusCode statusCode, out int bitValue)
+            {
+                int val = (int)statusCode;
+                Debug.Assert(val < 32 * BucketCount);
```

Comment:
What happens if OpenSSL releases a new version that we bind to and it ends up having > 96 errors defined?

Reply:
I added an overflow state, and verified behavior by having the shim report 94 / 95 / 96 as the value instead of CERT_EXPIRED